Warnings in sockets provider

As of this morning (18 Dec 2014), with gcc 4.9.1 on RHEL 6.5:

```
prov/sockets/src/sock_rma.c: In function 'sock_ep_rma_readv':
prov/sockets/src/sock_rma.c:186:20: warning: variable 'rma_iov' set but not used [-Wunused-but-set-variable]
  struct fi_rma_iov rma_iov;
                    ^
prov/sockets/src/sock_rma.c: In function 'sock_ep_rma_writev':
prov/sockets/src/sock_rma.c:349:20: warning: variable 'rma_iov' set but not used [-Wunused-but-set-variable]
  struct fi_rma_iov rma_iov;
                    ^
prov/sockets/src/sock_rma.c: In function 'sock_ep_rma_writedata':
prov/sockets/src/sock_rma.c:372:20: warning: variable 'rma_iov' set but not used [-Wunused-but-set-variable]
  struct fi_rma_iov rma_iov;
                    ^
prov/sockets/src/sock_rma.c: In function 'sock_ep_rma_inject':
prov/sockets/src/sock_rma.c:396:20: warning: variable 'rma_iov' set but not used [-Wunused-but-set-variable]
  struct fi_rma_iov rma_iov;
                    ^
prov/sockets/src/sock_rma.c: In function 'sock_ep_rma_injectdata':
prov/sockets/src/sock_rma.c:417:20: warning: variable 'rma_iov' set but not used [-Wunused-but-set-variable]
  struct fi_rma_iov rma_iov;
                    ^
```

--- a/prov/sockets/src/sock_progress.c
+++ b/prov/sockets/src/sock_progress.c
@@ -585,7 +585,7 @@ static int sock_pe_process_rx_read(struct sock_pe *pe, struct sock_rx_ctx *rx_ct
 static int sock_pe_process_rx_write(struct sock_pe *pe, struct sock_rx_ctx *rx_ctx,
 				   struct sock_pe_entry *pe_entry)
 {
-	int i, ret;
+	int i, ret = 0;
 	struct sock_mr *mr;
 	uint64_t offset, rem, len, entry_len, done_data, data_len;
 
@@ -1250,7 +1250,7 @@ out:
 static int sock_pe_process_rx_send(struct sock_pe *pe, struct sock_rx_ctx *rx_ctx,
 				   struct sock_pe_entry *pe_entry)
 {
-	ssize_t i, ret;
+	ssize_t i, ret = 0;
 	struct sock_rx_entry *rx_entry;
 	uint64_t len, rem, offset, data_len, done_data, used;
 

